File: OPENAPI-REST-API/swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/myadmin-client-csharp-dotnet2/Model/InlineResponse20012ServiceInfo.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class InlineResponse20012ServiceInfo {
    /// <summary>
    /// Gets or Sets ScrubIpId
    /// </summary>
    [DataMember(Name="scrub_ip_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_id")]
    public string ScrubIpId { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpType
    /// </summary>
    [DataMember(Name="scrub_ip_type", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_type")]
    public string ScrubIpType { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpCustid
    /// </summary>
    [DataMember(Name="scrub_ip_custid", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_custid")]
    public string ScrubIpCustid { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpOrderDate
    /// </summary>
    [DataMember(Name="scrub_ip_order_date", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_order_date")]
    public string ScrubIpOrderDate { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpIp
    /// </summary>
    [DataMember(Name="scrub_ip_ip", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_ip")]
    public string ScrubIpIp { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpServiceId
    /// </summary>
    [DataMember(Name="scrub_ip_service_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_service_id")]
    public string ScrubIpServiceId { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpServiceModule
    /// </summary>
    [DataMember(Name="scrub_ip_service_module", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_service_module")]
    public string ScrubIpServiceModule { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpStatus
    /// </summary>
    [DataMember(Name="scrub_ip_status", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_status")]
    public string ScrubIpStatus { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpInvoice
    /// </summary>
    [DataMember(Name="scrub_ip_invoice", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_invoice")]
    public string ScrubIpInvoice { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpCurrency
    /// </summary>
    [DataMember(Name="scrub_ip_currency", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_currency")]
    public string ScrubIpCurrency { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpCoupon
    /// </summary>
    [DataMember(Name="scrub_ip_coupon", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_coupon")]
    public string ScrubIpCoupon { get; set; }

    /// <summary>
    /// Gets or Sets ScrubIpComment
    /// </summary>
    [DataMember(Name="scrub_ip_comment", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "scrub_ip_comment")]
    public string ScrubIpComment { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class InlineResponse20012ServiceInfo {\n");
      sb.Append("  ScrubIpId: ").Append(ScrubIpId).Append("\n");
      sb.Append("  ScrubIpType: ").Append(ScrubIpType).Append("\n");
      sb.Append("  ScrubIpCustid: ").Append(ScrubIpCustid).Append("\n");
      sb.Append("  ScrubIpOrderDate: ").Append(ScrubIpOrderDate).Append("\n");
      sb.Append("  ScrubIpIp: ").Append(ScrubIpIp).Append("\n");
      sb.Append("  ScrubIpServiceId: ").Append(ScrubIpServiceId).Append("\n");
      sb.Append("  ScrubIpServiceModule: ").Append(ScrubIpServiceModule).Append("\n");
      sb.Append("  ScrubIpStatus: ").Append(ScrubIpStatus).Append("\n");
      sb.Append("  ScrubIpInvoice: ").Append(ScrubIpInvoice).Append("\n");
      sb.Append("  ScrubIpCurrency: ").Append(ScrubIpCurrency).Append("\n");
      sb.Append("  ScrubIpCoupon: ").Append(ScrubIpCoupon).Append("\n");
      sb.Append("  ScrubIpComment: ").Append(ScrubIpComment).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/InlineResponse20012ServiceInfo.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * InlineResponse20012ServiceInfo
 */
@Validated
@NotUndefined



public class InlineResponse20012ServiceInfo   {
  @JsonProperty("scrub_ip_id")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpId = null;

  @JsonProperty("scrub_ip_type")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpType = null;

  @JsonProperty("scrub_ip_custid")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpCustid = null;

  @JsonProperty("scrub_ip_order_date")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpOrderDate = null;

  @JsonProperty("scrub_ip_ip")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpIp = null;

  @JsonProperty("scrub_ip_service_id")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpServiceId = null;

  @JsonProperty("scrub_ip_service_module")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpServiceModule = null;

  /**
   * Gets or Sets scrubIpStatus
   */
  public enum ScrubIpStatusEnum {
    ACTIVE("active"),
    
    PENDING("pending"),
    
    CANCELED("canceled"),
    
    EXPIRED("expired");

    private String value;

    ScrubIpStatusEnum(String value) {
      this.value = value;
    }

    @Override
    @JsonValue
    public String toString() {
      return String.valueOf(value);
    }

    @JsonCreator
    public static ScrubIpStatusEnum fromValue(String text) {
      for (ScrubIpStatusEnum b : ScrubIpStatusEnum.values()) {
        if (String.valueOf(b.value).equals(text)) {
          return b;
        }
      }
      return null;
    }
  }
  @JsonProperty("scrub_ip_status")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private ScrubIpStatusEnum scrubIpStatus = null;

  @JsonProperty("scrub_ip_invoice")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpInvoice = null;

  @JsonProperty("scrub_ip_currency")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpCurrency = null;

  @JsonProperty("scrub_ip_coupon")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpCoupon = null;

  @JsonProperty("scrub_ip_comment")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String scrubIpComment = null;


  public InlineResponse20012ServiceInfo scrubIpId(String scrubIpId) { 

    this.scrubIpId = scrubIpId;
    return this;
  }

  /**
   * Get scrubIpId
   * @return scrubIpId
   **/
  
  @Schema(description = "")
  
  public String getScrubIpId() {  
    return scrubIpId;
  }



  public void setScrubIpId(String scrubIpId) { 
    this.scrubIpId = scrubIpId;
  }

  public InlineResponse20012ServiceInfo scrubIpType(String scrubIpType) { 

    this.scrubIpType = scrubIpType;
    return this;
  }

  /**
   * Get scrubIpType
   * @return scrubIpType
   **/
  
  @Schema(description = "")
  
  public String getScrubIpType() {  
    return scrubIpType;
  }



  public void setScrubIpType(String scrubIpType) { 
    this.scrubIpType = scrubIpType;
  }

  public InlineResponse20012ServiceInfo scrubIpCustid(String scrubIpCustid) { 

    this.scrubIpCustid = scrubIpCustid;
    return this;
  }

  /**
   * Get scrubIpCustid
   * @return scrubIpCustid
   **/
  
  @Schema(description = "")
  
  public String getScrubIpCustid() {  
    return scrubIpCustid;
  }



  public void setScrubIpCustid(String scrubIpCustid) { 
    this.scrubIpCustid = scrubIpCustid;
  }

  public InlineResponse20012ServiceInfo scrubIpOrderDate(String scrubIpOrderDate) { 

    this.scrubIpOrderDate = scrubIpOrderDate;
    return this;
  }

  /**
   * Get scrubIpOrderDate
   * @return scrubIpOrderDate
   **/
  
  @Schema(description = "")
  
  public String getScrubIpOrderDate() {  
    return scrubIpOrderDate;
  }



  public void setScrubIpOrderDate(String scrubIpOrderDate) { 
    this.scrubIpOrderDate = scrubIpOrderDate;
  }

  public InlineResponse20012ServiceInfo scrubIpIp(String scrubIpIp) { 

    this.scrubIpIp = scrubIpIp;
    return this;
  }

  /**
   * Get scrubIpIp
   * @return scrubIpIp
   **/
  
  @Schema(description = "")
  
  public String getScrubIpIp() {  
    return scrubIpIp;
  }



  public void setScrubIpIp(String scrubIpIp) { 
    this.scrubIpIp = scrubIpIp;
  }

  public InlineResponse20012ServiceInfo scrubIpServiceId(String scrubIpServiceId) { 

    this.scrubIpServiceId = scrubIpServiceId;
    return this;
  }

  /**
   * Get scrubIpServiceId
   * @return scrubIpServiceId
   **/
  
  @Schema(description = "")
  
  public String getScrubIpServiceId() {  
    return scrubIpServiceId;
  }



  public void setScrubIpServiceId(String scrubIpServiceId) { 
    this.scrubIpServiceId = scrubIpServiceId;
  }

  public InlineResponse20012ServiceInfo scrubIpServiceModule(String scrubIpServiceModule) { 

    this.scrubIpServiceModule = scrubIpServiceModule;
    return this;
  }

  /**
   * Get scrubIpServiceModule
   * @return scrubIpServiceModule
   **/
  
  @Schema(description = "")
  
  public String getScrubIpServiceModule() {  
    return scrubIpServiceModule;
  }



  public void setScrubIpServiceModule(String scrubIpServiceModule) { 
    this.scrubIpServiceModule = scrubIpServiceModule;
  }

  public InlineResponse20012ServiceInfo scrubIpStatus(ScrubIpStatusEnum scrubIpStatus) { 

    this.scrubIpStatus = scrubIpStatus;
    return this;
  }

  /**
   * Get scrubIpStatus
   * @return scrubIpStatus
   **/
  
  @Schema(description = "")
  
  public ScrubIpStatusEnum getScrubIpStatus() {  
    return scrubIpStatus;
  }



  public void setScrubIpStatus(ScrubIpStatusEnum scrubIpStatus) { 
    this.scrubIpStatus = scrubIpStatus;
  }

  public InlineResponse20012ServiceInfo scrubIpInvoice(String scrubIpInvoice) { 

    this.scrubIpInvoice = scrubIpInvoice;
    return this;
  }

  /**
   * Get scrubIpInvoice
   * @return scrubIpInvoice
   **/
  
  @Schema(description = "")
  
  public String getScrubIpInvoice() {  
    return scrubIpInvoice;
  }



  public void setScrubIpInvoice(String scrubIpInvoice) { 
    this.scrubIpInvoice = scrubIpInvoice;
  }

  public InlineResponse20012ServiceInfo scrubIpCurrency(String scrubIpCurrency) { 

    this.scrubIpCurrency = scrubIpCurrency;
    return this;
  }

  /**
   * Get scrubIpCurrency
   * @return scrubIpCurrency
   **/
  
  @Schema(description = "")
  
  public String getScrubIpCurrency() {  
    return scrubIpCurrency;
  }



  public void setScrubIpCurrency(String scrubIpCurrency) { 
    this.scrubIpCurrency = scrubIpCurrency;
  }

  public InlineResponse20012ServiceInfo scrubIpCoupon(String scrubIpCoupon) { 

    this.scrubIpCoupon = scrubIpCoupon;
    return this;
  }

  /**
   * Get scrubIpCoupon
   * @return scrubIpCoupon
   **/
  
  @Schema(description = "")
  
  public String getScrubIpCoupon() {  
    return scrubIpCoupon;
  }



  public void setScrubIpCoupon(String scrubIpCoupon) { 
    this.scrubIpCoupon = scrubIpCoupon;
  }

  public InlineResponse20012ServiceInfo scrubIpComment(String scrubIpComment) { 

    this.scrubIpComment = scrubIpComment;
    return this;
  }

  /**
   * Get scrubIpComment
   * @return scrubIpComment
   **/
  
  @Schema(description = "")
  
  public String getScrubIpComment() {  
    return scrubIpComment;
  }



  public void setScrubIpComment(String scrubIpComment) { 
    this.scrubIpComment = scrubIpComment;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    InlineResponse20012ServiceInfo inlineResponse20012ServiceInfo = (InlineResponse20012ServiceInfo) o;
    return Objects.equals(this.scrubIpId, inlineResponse20012ServiceInfo.scrubIpId) &&
        Objects.equals(this.scrubIpType, inlineResponse20012ServiceInfo.scrubIpType) &&
        Objects.equals(this.scrubIpCustid, inlineResponse20012ServiceInfo.scrubIpCustid) &&
        Objects.equals(this.scrubIpOrderDate, inlineResponse20012ServiceInfo.scrubIpOrderDate) &&
        Objects.equals(this.scrubIpIp, inlineResponse20012ServiceInfo.scrubIpIp) &&
        Objects.equals(this.scrubIpServiceId, inlineResponse20012ServiceInfo.scrubIpServiceId) &&
        Objects.equals(this.scrubIpServiceModule, inlineResponse20012ServiceInfo.scrubIpServiceModule) &&
        Objects.equals(this.scrubIpStatus, inlineResponse20012ServiceInfo.scrubIpStatus) &&
        Objects.equals(this.scrubIpInvoice, inlineResponse20012ServiceInfo.scrubIpInvoice) &&
        Objects.equals(this.scrubIpCurrency, inlineResponse20012ServiceInfo.scrubIpCurrency) &&
        Objects.equals(this.scrubIpCoupon, inlineResponse20012ServiceInfo.scrubIpCoupon) &&
        Objects.equals(this.scrubIpComment, inlineResponse20012ServiceInfo.scrubIpComment);
  }

  @Override
  public int hashCode() {
    return Objects.hash(scrubIpId, scrubIpType, scrubIpCustid, scrubIpOrderDate, scrubIpIp, scrubIpServiceId, scrubIpServiceModule, scrubIpStatus, scrubIpInvoice, scrubIpCurrency, scrubIpCoupon, scrubIpComment);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class InlineResponse20012ServiceInfo {\n");
    
    sb.append("    scrubIpId: ").append(toIndentedString(scrubIpId)).append("\n");
    sb.append("    scrubIpType: ").append(toIndentedString(scrubIpType)).append("\n");
    sb.append("    scrubIpCustid: ").append(toIndentedString(scrubIpCustid)).append("\n");
    sb.append("    scrubIpOrderDate: ").append(toIndentedString(scrubIpOrderDate)).append("\n");
    sb.append("    scrubIpIp: ").append(toIndentedString(scrubIpIp)).append("\n");
    sb.append("    scrubIpServiceId: ").append(toIndentedString(scrubIpServiceId)).append("\n");
    sb.append("    scrubIpServiceModule: ").append(toIndentedString(scrubIpServiceModule)).append("\n");
    sb.append("    scrubIpStatus: ").append(toIndentedString(scrubIpStatus)).append("\n");
    sb.append("    scrubIpInvoice: ").append(toIndentedString(scrubIpInvoice)).append("\n");
    sb.append("    scrubIpCurrency: ").append(toIndentedString(scrubIpCurrency)).append("\n");
    sb.append("    scrubIpCoupon: ").append(toIndentedString(scrubIpCoupon)).append("\n");
    sb.append("    scrubIpComment: ").append(toIndentedString(scrubIpComment)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
